compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -1238,16 +1238,20 @@ def FLOW_TensorTraceOp : FLOW_Op<"tensor.trace", []> {
 // Parameterization Ops
 //===---------------------------------------------------------------------===//
 
-def Flow_DispatchDefaultWorkgroupCountOp :
-    FLOW_PureOp<"dispatch.default_workgroup_count"> {
-  let summary = [{represents computation that returns workgroup count}];
+def Flow_DispatchWorkgroupCountFromDagRootOp :
+    FLOW_PureOp<"dispatch.workgroup_count_from_dag_root"> {
+  let summary = [{
+    workgroup count computed based on iteration range of the root of the DAG
+    for ops within dispatch
+  }];
   let description = [{
-    During dispatch region formation, the `workload` captured is used
-    by the backends to compute the number of workgroups. This operation
-    is a placeholder that represents the computation that takes the `workload`
-    as operands and returns the workgroup count as results.
-    The backends are expected to lower this operation into the actual computation
-    that returns the number of workgroups.
+    By default the IREEs backend relies on tile + distribution of the root
+    of the DAG of ops within the dispatch to split the work amongst workgroups.
+    So the workload captured is the size of the iteration space of the root
+    of the DAG. This op represents the computation that given the workload
+    returns the number of workgroups to use. The backends are responsible
+    for lowering this op into actual computation (typically based on the
+    tile sizes used to tile and distribute the root of the DAG).
```

Comment:
`workgroup_count_from_dag_root` is a pretty long name, and "computed based on iteration range of the root of the DAG for ops within dispatch" seems like it's leaking implementation details out?

Naming is hard, but maybe we can come up with a shorter (but still descriptive enough) name?

Reply:
naming is hard :P . When in doubt I bias towards more descriptive than less descriptive. Open to suggestions